src/module.c

```diff
@@ -3110,9 +3110,10 @@ int VM_ReplyWithErrorFormat(ValkeyModuleCtx *ctx, const char *fmt, ...) {
 int VM_ReplyWithSimpleString(ValkeyModuleCtx *ctx, const char *msg) {
     client *c = moduleGetReplyClient(ctx);
     if (c == NULL) return VALKEYMODULE_OK;
-    addReplyProto(c, "+", 1);
-    addReplyProto(c, msg, strlen(msg));
-    addReplyProto(c, "\r\n", 2);
+    if (prepareClientToWrite(c) != C_OK) return VALKEYMODULE_OK;
+    addReplyProto(c, "+", 1, 1);
+    addReplyProto(c, msg, strlen(msg), 1);
+    addReplyProto(c, "\r\n", 2, 1);
```

Comment:
I don't like delegating this internal detail to the caller.

Can we make a variadic version of addReplyProto so these three calls can be replaced with just one that does prepare client just once?

Reply:
Sure, I will update it.

Reply:
@zuiderkwast To keep this patch small and clean, I removed non related logic, only did optimization for `addReplyBulkCBuffer` which is in the critical path.
1.  Remove `addReplyLongLongWithPrefix` from header file, it is only used in `networking.c`.
2. Use `_addReplyToBufferOrList` to replace `addReplyProto` in `addReplyLongLongWithPrefix`, and move the prepare client logic to parent.